packages/wallet/test/services/ActiveStakePublicKeysTracker.test.ts

```diff
@@ -0,0 +1,159 @@
+import { AccountKeyDerivationPath, AsyncKeyAgent, GroupedAddress, KeyRole } from '@cardano-sdk/key-management';
+import { BehaviorSubject, firstValueFrom, from, take } from 'rxjs';
+import { Cardano } from '@cardano-sdk/core';
+import { ObservableWallet } from '../../src';
+import { createActivePublicStakeKeysTracker } from '../../src/services/ActiveStakePublicKeysTracker';
+import { mockProviders as mocks } from '@cardano-sdk/util-dev';
+
+describe('ActivePublicStakeKeysTracker', () => {
+  let addresses: GroupedAddress[];
+  let rewardAccounts: Cardano.RewardAccountInfo[];
+  let keyAgent: AsyncKeyAgent;
+  let derivePublicKey: jest.Mock;
+
+  /** Assert multiple emissions from stakePubKey$ */
+  const assertEmits = (
+    stakePubKeys$: ObservableWallet['activePublicStakeKeys$'],
+    emissions: string[][],
+    done: jest.DoneCallback
+    // eslint-disable-next-line unicorn/consistent-function-scoping
+  ) => {
+    const publicKeyEmissions: string[][] = [];
+    stakePubKeys$.pipe(take(emissions.length)).subscribe((v) => {
+      publicKeyEmissions.push(v);
+      if (publicKeyEmissions.length === 2) {
+        expect(publicKeyEmissions).toEqual(emissions);
+        done();
+      }
+    });
+  };
+
+  beforeEach(() => {
+    addresses = [
+      {
+        rewardAccount: mocks.rewardAccount,
+        stakeKeyDerivationPath: { index: 0, role: KeyRole.Stake }
+      },
+      {
+        rewardAccount: Cardano.RewardAccount('stake_test1upx9faamuf54pm7alg4lna5l7ll08pz833rj45tgr9m2jyceasqjt'),
+        stakeKeyDerivationPath: { index: 1, role: KeyRole.Stake }
+      }
+    ] as GroupedAddress[];
+
+    rewardAccounts = [
+      {
+        address: addresses[0].rewardAccount!,
+        keyStatus: Cardano.StakeKeyStatus.Registered,
+        rewardBalance: 1_000_000n
+      },
+      {
+        address: addresses[1].rewardAccount!,
+        keyStatus: Cardano.StakeKeyStatus.Registered,
+        rewardBalance: 1_000_000n
+      }
+    ];
+
+    derivePublicKey = jest
+      .fn()
+      .mockImplementation((path: AccountKeyDerivationPath) => Promise.resolve(`abc-${path.index}`));
+    keyAgent = {
+      derivePublicKey
+    } as unknown as AsyncKeyAgent;
+  });
+
+  it('empty array when there are no reward accounts', async () => {
+    const addresses$ = new BehaviorSubject<GroupedAddress[]>([]);
```

Comment:
Subjects look like overkill here. We would probably need to complete those as test cleanup too?
```suggestion
    const addresses$ = of<GroupedAddress[]>([]);
```

Reply:
Update in [938733f](https://github.com/input-output-hk/cardano-js-sdk/commit/938733fec40d442887ca83f1643f80e53a019b1a)